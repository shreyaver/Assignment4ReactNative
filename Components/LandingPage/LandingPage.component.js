/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import React, { Component } from 'react';
import {
  SafeAreaView, Text, TouchableOpacity, Platform, ScrollView, View,
} from 'react-native';
import axios from 'axios';
import PropTypes from 'prop-types';
import styles from './LandingPage.component.stylesheet';
import Icon from "react-native-vector-icons/Entypo";
import { GET_FORMS_URL_IOS, GET_FORMS_URL_ANDROID } from '../../config';
import FormCard from '../FormCard/FormCard.component';

export default class LandingPage extends Component {
  static navigationOptions = {
    title: 'Home',
  };
  
  state = { forms: []};

  async componentDidMount() {
    try {
      const forms = await axios.get(Platform.OS === 'ios' ? GET_FORMS_URL_IOS : GET_FORMS_URL_ANDROID);
      this.setState({ forms: forms.data });
    } catch (errorObj) {
      console.log(errorObj);
    }
  }

  render() {
    const formCards = this.state.forms.map((form, index) => (<FormCard form={form} key={index} />));
    const { navigation } = this.props;
    return (
      <SafeAreaView style={styles.safeAreaView}>
          <View style={styles.header}>
            <Text style={styles.heading}> Awesome Forms </Text>   
            <Icon
              name="dots-three-vertical"
              color="white"
              size={40}
              style={styles.dotIcon}
            />
          </View>       
          <ScrollView contentContainerStyle={styles.screenContent}>
            {formCards}
          </ScrollView>
          <TouchableOpacity style={styles.createFormButton} testID="navToNewForm" onPress={() => navigation.navigate('NewForm')}>
              <Icon                            
                name="plus"
                size={30}      
                color="white"      
              />    
          </TouchableOpacity>
      </SafeAreaView>
    );
  }
}
LandingPage.propTypes = {
  navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired }).isRequired,
};