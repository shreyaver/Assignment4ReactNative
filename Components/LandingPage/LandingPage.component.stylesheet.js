import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  safeAreaView: {
    flex: 1,
    backgroundColor: '#3F51B5',
  },
  header: {
    height: 90,
    width: '100%',
    backgroundColor: '#3F51B5',
    flexDirection: 'row',
    alignItems: 'center',
  },
  heading: {
    color: 'white',
    fontSize: 22,
    marginLeft: 20,
  },
  dotIcon: {
    marginLeft: 'auto',
  },
  createFormButton: {
    height: 70,
    width: 70,
    borderRadius: 35,
    justifyContent: 'center',
    position: 'absolute',
    bottom: 40,
    right: 40,
    alignItems: 'center',
    backgroundColor: '#3F51B5',
  },
  screenContent: {
    width: '100%',
    flexGrow: 1,
    paddingTop: 30,
    backgroundColor: '#D8E4F0',
    alignItems: 'center',
  },
  cardList: {
    width: '80%',
  },
});
