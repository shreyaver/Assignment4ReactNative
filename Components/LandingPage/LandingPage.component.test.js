import React from 'react';
import Axios from 'axios';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LandingPage from './LandingPage.component';

describe('LandingPage', () => {
  let mockGetForms;
  let navigation;
  const mockFormData = [
    {
      name: 'Personal Details',
      createdAt: "2019-01-01T12:00:00.000Z",
    }, {
      name: 'Code Academy Feedback 2019',
      createdAt: "2019-01-01T12:00:00.000Z",
    },
  ]
  beforeAll(() => {
    mockGetForms = jest.spyOn(Axios, 'get');
    mockGetForms.mockImplementation(() => Promise.resolve({ data: mockFormData }));
    navigation = { navigate: jest.fn() };
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });
  it('renders without crashing', () => {
    const landingPage = renderer.create(<LandingPage navigation={navigation} />).toJSON();
    expect(landingPage).toMatchSnapshot();
  });
  it('sets initial state to empty array for "forms"', () => {
    const wrapper = shallow(<LandingPage navigation={navigation} />);
    expect(wrapper.instance().state).toEqual({ forms: [] });
  });

  it('sets state to fetched data after mounting', async () => {
    const wrapper = shallow(<LandingPage navigation={navigation} />);
    await wrapper.instance().componentDidMount();
    expect(wrapper.instance().state).toEqual({ forms: mockFormData });
  });

  it('directs to "NewForm" page on clicking button in header', () => {
    const wrapper = shallow(<LandingPage navigation={navigation} />);
    expect(navigation.navigate).not.toHaveBeenCalled();
    wrapper.find({ testID: 'navToNewForm' }).simulate('press');
    expect(navigation.navigate).toHaveBeenCalledWith('NewForm');
  });
});