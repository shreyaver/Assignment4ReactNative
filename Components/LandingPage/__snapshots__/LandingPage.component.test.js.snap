// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders without crashing 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#3F51B5",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3F51B5",
        "flexDirection": "row",
        "height": 90,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 20,
        }
      }
    >
       Awesome Forms 
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 40,
          },
          Object {
            "marginLeft": "auto",
          },
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D8E4F0",
        "flexGrow": 1,
        "paddingTop": 30,
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3F51B5",
        "borderRadius": 35,
        "bottom": 40,
        "height": 70,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 40,
        "width": 70,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</RCTSafeAreaView>
`;
